File: workshop/python/ex03_pure_vs_io/before.py
def process_orders(orders):
    total = 0
    for o in orders:
        if o.get("status") == "cancelled":
            print("skip", o.get("id"))
            continue
        qty = int(o.get("qty", 0))
        price = float(o.get("price", 0))
        subtotal = qty * price
        if subtotal > 1000:
            subtotal *= 0.95
        total += subtotal
        print("order", o.get("id"), "subtotal", subtotal)
    print("total", total)
    return round(total, 2)

File: workshop/node/ex01_pure_vs_io/before.js
function processOrders(orders) {
    let total = 0;
    for (const o of orders) {
    if (o.status === "cancelled") {
        console.log("skip", o.id);
        continue;
    }
    const qty = parseInt(o.qty ?? 0, 10);
    const price = parseFloat(o.price ?? 0);
    let subtotal = qty * price;
    if (subtotal > 1000) subtotal *= 0.95;
    total += subtotal;
    console.log("order", o.id, "subtotal", subtotal);
    }
    console.log("total", total);
  return Math.round(total * 100) / 100;
}

module.exports = { processOrders };